### PR TITLE
doc: matter: update existing docs for Wi-Fi

### PR DESCRIPTION
**Pull request by greg-fer**
Update existing Matter docs about Wi-Fi.
KRKNWK-14783.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_matter_configuring.rst [hunk 1]
```diff
@@ -29,14 +25,30 @@ The following figure shows the protocol types and the available Matter controlle
...
+       <div style="text-align: center;">
+       <iframe width="560" height="315" src="https://www.youtube.com/embed/XWWJYraNZPs" title="Matter development environment video overview" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+       </div>
+
+
+  * To enable the IPv6 communication with the Matter accessory device over the Wi-Fi network, no additional configuration is required, because the Matter controller can interact with any Wi-Fi device directly and no additional bridging is needed.
+
+* Matter controller setup:
+
+  * You can run it on the same device as the Thread Border Router or Wi-Fi device.
```
Comment (kkasperczyk-no) [inline doc/nrf/ug_matter_configuring.rst:48]: What is the Wi-Fi device in this setup? The Matter over Wi-Fi accessory? If so I would not say it's true, as those rather should be separate devices.

File: doc/nrf/ug_matter_configuring.rst [hunk 2]
```diff
@@ -6,19 +6,15 @@ Configuring Matter
...
-Currently, the |NCS| only supports Matter stack that is built on top of a low-power, 802.15.4-compatible Thread network.
-To commission and control the Matter accessory device remotely over such a network, you need to set up a Matter controller on a mobile or a PC.
+Currently, the |NCS| only supports Matter stack that is built on top of a low-power, 802.15.4-compatible Thread network and over the Wi-Fi network.
+To commission and control the Matter accessory device remotely over either of these networks, you need to set up a Matter controller on a mobile or a PC.
 Matter controller is a role within the Matter environment.
 This controller interacts with the accessory devices using the following protocols:
 
 * Bluetooth® LE during the commissioning process - to securely pass the network credentials and provision the accessory device into the Thread network.
-* Regular IPv6 communication after the accessory device joins the Thread network - to interact with each other by exchanging application messages.
+* Regular IPv6 communication after the accessory device joins the IPv6 network - to interact with each other by exchanging application messages.
```
Comment (markaj-nordic) [inline doc/nrf/ug_matter_configuring.rst:17]:
```suggestion
* Regular IPv6 communication after the accessory device joins the Thread or WiFi network - to interact with each other by exchanging application messages.
```

File: doc/nrf/ug_matter_configuring_controller.rst [hunk 3]
```diff
@@ -54,7 +54,7 @@ CHIP Tool for Android
 *********************
 
 CHIP Tool for Android (also known as Android CHIPTool) is the recommended Mobile controller for mobile, which allows you to test Matter applications using an Android smartphone.
-You can use it in the :ref:`development environment setup option <ug_matter_configuring>` where Thread Border Router and Matter controller are running on separate devices.
+You can use it in the :ref:`development environment setup option <ug_matter_configuring>` where the IPv6 network device and Matter controller are running on separate devices.
```
Comment (kkasperczyk-no) [inline doc/nrf/ug_matter_configuring_controller.rst:57]: This is similar comment, as above. In setup with Thread we've got additional device that is Thread Border Router, but for Wi-Fi there isn't corresponding device for it, so I'm not sure what is IPv6 network device here.

File: doc/nrf/ug_matter_configuring.rst [hunk 4]
```diff
@@ -29,14 +25,30 @@ The following figure shows the protocol types and the available Matter controlle
...
 
-Matter development environment setup options depend on how you run the Matter controller.
-You can either run it on the same device as the Thread Border Router or run the Matter controller and the Thread Border Router on separate devices.
+* Network type:
 
-Depending on your choice, you can set up one of the following development environments:
+  * To enable the IPv6 communication with the Matter accessory device over the Thread network, the Matter controller requires the Thread Border Router.
+    This is because the Matter controller types described on this page do not have an 802.15.4 Thread interface.
+    The Border Router bridges the Thread network with the network interface of the controller, for example Wi-Fi.
+    Watch the following video for an uoverview of the Matter over Thread setup.
```
Comment (markaj-nordic) [inline doc/nrf/ug_matter_configuring.rst:35]:
```suggestion
    Watch the following video for an overview of the Matter over Thread setup.
```

File: doc/nrf/ug_matter_configuring.rst [hunk 5]
```diff
@@ -29,14 +25,30 @@ The following figure shows the protocol types and the available Matter controlle
...
+
+    .. raw:: html
+
+       <div style="text-align: center;">
+       <iframe width="560" height="315" src="https://www.youtube.com/embed/XWWJYraNZPs" title="Matter development environment video overview" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+       </div>
+
+
+  * To enable the IPv6 communication with the Matter accessory device over the Wi-Fi network, no additional configuration is required, because the Matter controller can interact with any Wi-Fi device directly and no additional bridging is needed.
```
Comment (markaj-nordic) [inline doc/nrf/ug_matter_configuring.rst:44]: Well the access point is needed to be clear, and Matter controller and a node must be in the same network

File: doc/nrf/ug_matter_architecture.rst [hunk 6]
```diff
@@ -22,8 +22,9 @@ That is, the code used for the |NCS| and Matter integration is stored in the Mat
 Both instances depend on each other, but their development is independent to ensure that they both support the latest stable version of one another.
 
 Matter is located on the top application layer of the integration model, looking from the networking point of view.
-The |NCS| and Zephyr provide the Bluetooth® LE and Thread stacks, which must be integrated with the Matter stack using a special intermediate layer.
-The |NCS|'s Multiprotocol Service Layer (MPSL) driver allows running Bluetooth LE and Thread concurrently on the same radio chip.
+The |NCS| and Zephyr provide the Bluetooth® LE, Thread, and Wi-Fi stacks, which must be integrated with the Matter stack using a special intermediate layer.
+Depending on the selected transport, either Wi-Fi or Thread is enabled.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_architecture.rst:26]:
```suggestion
Depending on the selected build target, either Wi-Fi or Thread is enabled.
```

File: doc/nrf/ug_matter_configuring_env.rst [hunk 7]
```diff
@@ -11,19 +11,19 @@ Setting up Matter development environment is all about setting up the Matter net
...
-As a result, the device can join the Thread network and communicate with other Thread devices in the network.
+During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the IPv6 network credentials to the Matter accessory device.
+As a result, the device can join the IPv6 network and communicate with other IPv6 devices in the network.
 
 To start the commissioning procedure, the controller must get the onboarding information from the Matter accessory device.
 The data payload, which includes the device discriminator and setup PIN code, is encoded within a QR code, printed to the UART console, and can be shared using an NFC tag.
 
 Matter development environment setup depends on how you choose to run the Matter controller.
-You can either run it on the same device as the Thread Border Router or run the Matter controller and the Thread Border Router on separate devices.
+You can either run it on the same IPv6 network device or run it on a separate device.
```
Comment (markaj-nordic) [inline doc/nrf/ug_matter_configuring_env.rst:21]: IMO it's not clear, i.e. what we mean by IPv6 network device.

File: doc/nrf/ug_matter_configuring.rst [hunk 8]
```diff
@@ -6,19 +6,15 @@ Configuring Matter
...
 
-.. raw:: html
 
-   <div style="text-align: center;">
-   <iframe width="560" height="315" src="https://www.youtube.com/embed/XWWJYraNZPs" title="Matter development environment video overview" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
-   </div>
 
-Currently, the |NCS| only supports Matter stack that is built on top of a low-power, 802.15.4-compatible Thread network.
-To commission and control the Matter accessory device remotely over such a network, you need to set up a Matter controller on a mobile or a PC.
+Currently, the |NCS| only supports Matter stack that is built on top of a low-power, 802.15.4-compatible Thread network and over the Wi-Fi network.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:11]:
```suggestion
The |NCS| supports Matter stack that is built on top of a low-power, 802.15.4-compatible Thread network, or over the Wi-Fi network.
```
We now support 2 out of 3 networking technologies covered by Matter, so we no longer need to be shy and say "only" :)

File: samples/matter/lock/Kconfig [hunk 9]
```diff
@@ -26,6 +26,23 @@ config STATE_LEDS
 	  the device into a network or the factory reset initiation. Note that setting this option to
 	  'n' does not disable the LED indicating the state of the simulated bolt.
 
+# Sample configuration used for Thread networking
```
Comment (markaj-nordic) [inline samples/matter/lock/Kconfig:29]: ?? Rebase?

File: doc/nrf/ug_matter_configuring.rst (hunk 4 above)
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:35]:
```suggestion
    Watch the following video for an overview of the Matter over Thread setup.
```

File: doc/nrf/ug_matter_configuring.rst (hunk 1 above)
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:48]: We do not describe any setup where the controller is run on the Wi-Fi device, do we?

**Review by markaj-nordic — COMMENTED** [review]
Brief review.

File: doc/nrf/ug_matter_configuring_controller.rst [hunk 10]
```diff
@@ -36,7 +36,7 @@ CHIP Tool for Linux or macOS
 ****************************
 
 CHIP Tool for Linux or macOS is the default implementation of the Matter controller role, recommended for the nRF Connect platform.
-You can use it either on a separate device from Thread Border Router or on the same device as Thread Border Router, depending on which :ref:`development environment setup option <ug_matter_configuring>` you choose.
+You can use it either on a device connected to the IPv6 network or on a separate device, depending on which :ref:`development environment setup option <ug_matter_configuring>` you choose.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring_controller.rst:39]: The controller must be run on a device connected to the IPv6 network. The only options we have is when testing a Thread device - then we can run the controller on the same device as OTBR or on a separate one. I would leave the previous sentence and only add info that this option applies only to testing Thread devices.

File: doc/nrf/ug_matter_configuring_env.rst [hunk 11]
```diff
@@ -11,19 +11,19 @@ Setting up Matter development environment is all about setting up the Matter net
 In Matter, the commissioning procedure takes place over Bluetooth LE between a Matter accessory device and the Matter controller, where the controller has the commissioner role.
 When the procedure has completed, the device should be equipped with all information needed to securely operate in the Matter network.
 
-During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the Thread network credentials to the Matter accessory device.
-As a result, the device can join the Thread network and communicate with other Thread devices in the network.
+During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the IPv6 network credentials to the Matter accessory device.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring_env.rst:14]:
```suggestion
During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the Thread or WiFi network credentials to the Matter accessory device.
```
While both Thread and WiFi are technologies that IPv6 can use, the security component only exists at the underlying layer, so "IPv6 network credentials" term may be confusing.

File: samples/matter/lock/Kconfig (hunk 9 above)
Comment (greg-fer) [reply samples/matter/lock/Kconfig:29]: Yes

File: doc/nrf/ug_matter_configuring.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring.rst]: Now we do!

File: doc/nrf/ug_matter_configuring.rst [hunk 12]
```diff
@@ -29,37 +24,34 @@ The following figure shows the protocol types and the available Matter controlle
...
-     :start-after: matter_env_ctrl_pc_start
-     :end-before: matter_env_ctrl_pc_end
 
-* Thread Border Router on Raspberry Pi and Matter controller for mobile (CHIP Tool for Android)
+  * To enable the IPv6 communication with the Matter accessory device over the Wi-Fi network, you need a Wi-Fi Access Point, for example a Wi-Fi router, so that the Matter controller can interact with any Wi-Fi device directly.
 
-  .. include:: /ug_matter_configuring_env.rst
-     :start-after: matter_env_ctrl_mobile_start
-     :end-before: matter_env_ctrl_mobile_end
+* Matter controller setup:
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:45]: The entire choice is for Matter over Thread only. In Wifi you don't have the presented choice (to run it on TBR or separate device). Using the word `separate` alone without telling `from what` seems confusing to me.

File: doc/nrf/ug_matter_gs_kconfig.rst [hunk 13]
```diff
@@ -0,0 +1,24 @@
+.. _ug_matter_gs_kconfig:
+.. _ug_matter_configuring_protocol:
+
+Enabling Matter in Kconfig
+##########################
+
+To use the Matter protocol, set the following Kconfig options:
+
+* :kconfig:option:`CONFIG_CHIP` - This option enables the Matter protocol stack and other associated Kconfig options, including :kconfig:option:`CONFIG_CHIP_ENABLE_DNSSD_SRP` that is required for the discovery of the Matter device using DNS-SD.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_gs_kconfig.rst:9]:
```suggestion
* :kconfig:option:`CONFIG_CHIP` - This option enables the Matter protocol stack and other associated features and components, such as C++ support or Zephyr networking layer.
```
SRP is for Thread only, so better to use transport-agnostic examples.

File: doc/nrf/ug_matter_gs_kconfig.rst [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+* :kconfig:option:`CONFIG_CHIP_PROJECT_CONFIG` - This option defines the path to the configuration file that specifies Vendor ID, Product ID, and other project-specific Matter settings.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_gs_kconfig.rst:10]: Currently, most settings that used to be configured with this file can be configured via Kconfig, so there's no need to mention it here. I would move it to the advanced configuration part with description: "This option defines the path to the configuration file that contains project-specific Matter settings in the form of C preprocessor macros, that cannot be altered using dedicated Kconfig options"

File: samples/matter/light_bulb/README.rst [hunk 15]
```diff
@@ -50,6 +50,7 @@ Remote testing in a network
 
 By default, the Matter accessory device has Thread disabled.
 You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread network.
+You have to make the device discoverable manually (for security reasons).
```
Comment (Damian-Nordic) [inline samples/matter/light_bulb/README.rst:53]: Nope, please remove.

**Review by Damian-Nordic — APPROVED** [review]
Some remaining comments, but lgtm

**Comment by greg-fer** [comment]
See https://github.com/nrfconnect/sdk-nrf/pull/8474

**Pull request closed without merging**